pysteps/tracking/tdating.py

```diff
@@ -0,0 +1,309 @@
+
+References
+...............
+:cite:`TRT2004`
+
+@author: mfeldman
+
+.. autosummary::
+    :toctree: ../generated/
+
+    dating
+    tracking
+    advect
+    match
+    couple_track
+"""
+
+import copy
+import sys
+
+import numpy as np
+import scipy.ndimage as ndi
+
+import pysteps.feature.tstorm as tstorm_detect
+from pysteps import motion
+from pysteps.exceptions import MissingOptionalDependency
+
+try:
+    import skimage
+
+    SKIMAGE_IMPORTED = True
+except ImportError:
+    SKIMAGE_IMPORTED = False
+if SKIMAGE_IMPORTED:
+    import skimage.draw as skid
+    import skimage.feature as skif
+    import skimage.measure as skime
+    import skimage.morphology as skim
+    import skimage.segmentation as skis
+try:
+    import pandas as pd
+
+    PANDAS_IMPORTED = True
+except ImportError:
+    PANDAS_IMPORTED = False
+
+
+def dating(
+    input_video,
+    timelist,
+    mintrack=3,
+    cell_list=None,
+    label_list=None,
+    start=0,
+    minref=35,
+    maxref=48,
+    mindiff=6,
+    minsize=50,
+    minmax=41,
+    mindis=10,
+    dyn_thresh=False,
+):
+    """
+    This function performs the thunderstorm detection and tracking DATing.
+    It requires a 3-D input array that contains the temporal succession of the 2-D data
+    array of each timestep. On each timestep the detection is performed, the identified
+    objects are advected with a flow prediction and the advected objects are matched to
+    the newly identified objects of the next timestep.
+    The last portion re-arranges the data into tracks sorted by ID-number.
+
+    Parameters
+    ----------
+    input_video : array-like
+        Array of shape (t,m,n) containing input image, with t being the temporal
+        dimension and m,n the spatial dimensions. Nan values are ignored.
+    timelist : list
+        List of length t containing string of time and date of each (m,n) field.
+    mintrack : int, optional
+        minimum length of cell-track to be counted. The default is 3.
+    cell_list : list or None, optional
+        If you wish to expand an existing list of cells, insert previous cell-list here.
+        The default is None.
+        If not None, requires that label_list has the same length.
+    label_list : list or None, optional
+        If you wish to expand an existing list of cells, insert previous label-list here.
+        The default is None.
+        If not None, requires that cell_list has the same length.
+    start : int, optional
+        If you wish to expand an existing list of cells, the input video must contain 2
+        timesteps prior to the merging. The start can then be set to 2, allowing the
+        motion vectors to be formed from the first three grids and continuing the cell
+        tracking from there. The default is 0, which initiates a new tracking sequence.
+    minref : float, optional
+        Lower threshold for object detection. Lower values will be set to NaN.
+        The default is 35 dBZ.
```

Comment:
This might confuse some users. Default values are specified in dBZ. However, the expected unit of the input data is not specified anywhere. I guess it can take any input field regardless of the unit?

Reply:
To me it looks like minref (and several other keyword arguments) are not used anywhere in the code.

Reply:
I expanded the description of the default input.
Thank you very much for pointing out the lack of use of the keywords. This was a mistake in the code, that has now been corrected. Unfortunately this revealed some ongoing issues in the dynthresh function, so I removed it for now.